components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/SSOLoginReCaptchaConfig.java

```diff
@@ -130,18 +141,26 @@ public CaptchaPreValidationResponse preValidate(ServletRequest servletRequest, S
             preValidationResponse.setCaptchaAttributes(params);
             preValidationResponse.setOnCaptchaFailRedirectUrls(CaptchaUtil.getOnFailedLoginUrls());
             preValidationResponse.setCaptchaValidationRequired(true);
-
-        } else if (CaptchaUtil.isMaximumFailedLoginAttemptsReached(MultitenantUtils.getTenantAwareUsername(username),
-                tenantDomain)) {
-            preValidationResponse.setCaptchaValidationRequired(true);
-            preValidationResponse.setMaxFailedLimitReached(true);
-
-            preValidationResponse.setOnCaptchaFailRedirectUrls(CaptchaUtil.getOnFailedLoginUrls());
-            Map<String, String> params = new HashMap<>();
-            params.put(CaptchaConstants.RE_CAPTCHA, CaptchaConstants.TRUE);
-            params.put(CaptchaConstants.AUTH_FAILURE, CaptchaConstants.TRUE);
-            params.put(CaptchaConstants.AUTH_FAILURE_MSG, CaptchaConstants.RECAPTCHA_FAIL_MSG_KEY);
-            preValidationResponse.setCaptchaAttributes(params);
+        } else {
+            String failedLoginAttemptsClaimUri = FAILED_LOGIN_ATTEMPTS_CLAIM_URI;
+            if (EMAIL_OTP_AUTHENTICATOR_NAME.equals(context.getCurrentAuthenticator())) {
+                failedLoginAttemptsClaimUri = FAILED_EMAIL_OTP_ATTEMPTS_CLAIM_URI;
+            }
+            if (SMS_OTP_AUTHENTICATOR_NAME.equals(context.getCurrentAuthenticator())) {
+                failedLoginAttemptsClaimUri = FAILED_SMS_OTP_ATTEMPTS_CLAIM_URI;
+            }
```

Comment:
This introduces a dependency to these 2 specific authenticators, right?

Reply:
Yes, should be aware of the names of the authenticators.